client/src/Picks.tsx

```diff
@@ -1020,20 +1047,30 @@ function DescriptionHistory({ author, edits }: DescriptionHistoryProps) {
       <Modal close={() => setModalOpen(false)} open={modalOpen}>
         <h2>Description history</h2>
         <div className='description-history'>
-          {edits.map((edit) => (
-            <div key={edit.id} className='description-history-item'>
-              <h3 className='description-history-item__title'>
-                Edit by <UserInline user={edit.editor} />
-                {edit.editor_id === author?.id && ' (author)'} on{' '}
-                <FormattedDate dateStyle='long' timeStyle='medium' value={edit.edited_at} />
-              </h3>
-              {edit.description == null ? (
-                <i>No description</i>
-              ) : (
-                <BBCode text={edit.description} />
-              )}
-            </div>
-          ))}
+          {edits.map((edit, index) => {
+            const previousDescription =
+              index === 0 ? edit.description : edits[index - 1].description;
+
+            return (
+              <div key={edit.id} className='description-history-item'>
+                <h3 className='description-history-item__title'>
+                  Edit by <UserInline user={edit.editor} />
+                  {edit.editor_id === author?.id && ' (author)'} on{' '}
+                  <FormattedDate dateStyle='long' timeStyle='medium' value={edit.edited_at} />
+                </h3>
+                {edit.description == null ? (
+                  <i>No description</i>
+                ) : (
+                  <BBCode
+                    text={visualizeDescriptionDifferences(
+                      previousDescription ?? edit.description,
+                      edit.description,
+                    )}
+                  />
```

Comment:
this doesn't make sense, you're taking the diff of bbcode and expecting the result to be valid bbcode
either diff bbcode and display without parsing it, or parse bbcode first and diff the syntax trees (this diff library won't work with latter though)

Reply:
both the library you're using now and that one should work fine if you don't render out bbcode at the end i.e. don't use `BBCode` component. i just can't tell if u were going for that or wanted to make something like "rich diff" that actually considers the formatting as part of the diff